data/mods/gen9ssb/moves.ts

```diff
@@ -275,6 +275,45 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 		type: "Grass",
 	},
 
+	// Arya
+	anyonecanbekilled: {
+		accuracy: 95,
+		basePower: 80,
+		category: "Status",
+		shortDesc: "Raises the user's Sp. Atk by 2 stages for the next 2 turns, -2 Sp. Atk afterwards",
+		name: "Anyone can be killed",
+		pp: 15,
+		priority: 0,
+		flags: {sound: 1},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		volatileStatus: 'anyonecanbekilled',
+		condition: {
+			duration: 3,
+			onResidualOrder: 3,
+			onStart(target, source, sourceEffect) {
+				this.boost({spa: 2}, source);
+			},
+			onEnd(target) {
+				const activePokemon = this.getAllActive();
+				for (const a of activePokemon) {
+					if (a.name === "Arya" && a.side !== target.side) {
+						this.boost({spa: -2}, a);
+						break;
+					}
+				}
+			},
```

Comment:
you dont need a for loop here, since onEnd takes a pokemon arg (pokemon being the volatile holder)

Reply:
that's the problem though, i checked this during testing. removing the loop and having just the debuff in onEnd applies it to the opponent, not the user.

Reply:
On second looksee, thats because volatileStatus is being applied to the opponent, not the user